### PR TITLE
Jamie/angular 9 followup fixes 2

### PR DESCRIPTION
**Pull request by SEAjamieD**
### :nut_and_bolt: Description: What code changed, and why?
This branch adds a few more fixes for type checks in our new angular 9 build

### :+1: Definition of Done
Build comes out clean

### :athletic_shoe: How to Build and Test the Change
build components/automate-ui-devproxy
make serve

### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] Tests added/updated?
- [ ] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: components/automate-ui/src/app/pages/+compliance/shared/reporting/report-data.service.ts [hunk 1]
```diff
@@ -9,7 +9,10 @@ interface ReportingSummary {
     nodes: any;
     profiles: any;
     controls?: any;
+    platforms: number;
+    environments: number;
   };
+  status: 'failed' | 'passed' | 'waived' | 'skipped' | 'unknown';
```
Comment (msorens) [inline components/automate-ui/src/app/pages/+compliance/shared/reporting/report-data.service.ts]: If you have spare cycles, would be better to make this type a reusable string literal type, so you can then strongly type other things that use it, specifically `getIcon` and `formatSummaryPhrase` in reporting.component.ts.

Comment (SEAjamieD) [reply components/automate-ui/src/app/pages/+compliance/shared/reporting/report-data.service.ts]: There is no model file set up importing types for reporting.component.ts, would it be strange to create the interface for status in this file here (report-data.service.ts) and import it into reporting-component.ts?

Comment (SEAjamieD) [reply components/automate-ui/src/app/pages/+compliance/shared/reporting/report-data.service.ts]: I added what I thought made the most sense.

File: components/automate-ui/src/app/pages/+compliance/+reporting/reporting.component.ts [hunk 2]
```diff
@@ -411,7 +412,7 @@ export class ReportingComponent implements OnInit, OnDestroy {
     this.showSummary = !this.showSummary;
   }
 
-  getIcon(status) {
+  getIcon(status: ReportingSummaryStatus) {
```
Comment (msorens) [inline components/automate-ui/src/app/pages/+compliance/+reporting/reporting.component.ts]:
```suggestion
  getIcon(status: ReportingSummaryStatus): string {
```

File: components/automate-ui/src/app/pages/+compliance/+reporting/reporting.component.ts [hunk 3]
```diff
@@ -421,7 +422,7 @@ export class ReportingComponent implements OnInit, OnDestroy {
     }
   }
 
-  formatSummaryPhrase(status) {
+  formatSummaryPhrase(status: ReportingSummaryStatus) {
```
Comment (msorens) [inline components/automate-ui/src/app/pages/+compliance/+reporting/reporting.component.ts]:
```suggestion
  formatSummaryPhrase(status: ReportingSummaryStatus): string {
```

File: components/automate-ui/src/app/pages/+compliance/+reporting/reporting.component.ts [hunk 4]
```diff
@@ -30,6 +30,7 @@ import {
 import { DateTime } from 'app/helpers/datetime/datetime';
 import { pickBy } from 'lodash/fp';
 import { FilterC } from './types';
+import { ReportingSummaryStatus } from '../shared/reporting/report-data.service';
```
Comment (msorens) [inline components/automate-ui/src/app/pages/+compliance/+reporting/reporting.component.ts]: Delete this line and add `ReportingSummaryStatus` to the import on L18.

**Pull request merged**